From be9ade9950 Mon Sep 17 00:00:00 2001
From: Nathan Scott <nathans@redhat.com>
Subject: [PATCH] qa/src/scanmeta.c: fix call to __pmLogLoadInDom()

scanmeta was *using* the acp == NULL guard to dodge the rlen test and
calling with rlen == 0 (this QA app was simply assuming the record
was valid, at least to the point that the buffer could be correctly
decoded).

Fix involves re-extracting the correct record length and calling
__pmLogLoadInDom() with rlen != 0.
---
diff --git a/qa/src/scanmeta.c b/qa/src/scanmeta.c
--- a/qa/src/scanmeta.c
+++ b/qa/src/scanmeta.c
@@ -143,7 +143,7 @@
 }
 
 void
-do_indom(__int32_t *buf, int type)
+do_indom(__int32_t *buf, int type, int len)
 {
     int				sts;
     static __pmTimestamp	prior_stamp = { 0, 0 };
@@ -156,7 +156,7 @@
     elt_t			*tp;
     elt_t			*dp = &dup;
 
-    if ((sts = __pmLogLoadInDom(NULL, 0, type, &lid, &buf)) < 0) {
+    if ((sts = __pmLogLoadInDom(NULL, len, type, &lid, &buf)) < 0) {
 	fprintf(stderr, "__pmLoadLoadInDom: failed: %s\n", pmErrStr(sts));
 	return;
     }
@@ -689,7 +689,7 @@
 	    case TYPE_INDOM_V2:
 		if (!iflag)
 		    break;
-		do_indom(buf, hdr.type);
+		do_indom(buf, hdr.type, hdr.len);
 		break;
 
 	    case TYPE_LABEL:
